### PR TITLE
Port to Caddy 0.9

### PR DESCRIPTION
**Pull request by Merccy**

### CODE REVIEW

**Comment by pyed** [comment]
What about the tests ?

**Comment by Merccy** [comment]
@pyed Sorry, forgot about them. They have now been updated.

**Comment by pyed** [comment]
use `git rebase` to squash them into one commit.

**Comment by Merccy** [comment]
Done

**Comment by pyed** [comment]
Thank you for your contribution

File: ipfilter.go [hunk 1]
```diff
@@ -89,19 +89,33 @@ func block(blockPage string, w *http.ResponseWriter) (int, error) {
...
+	// Create new middleware
+	newMiddleWare := func(next httpserver.Handler) httpserver.Handler {
 		return &IPFilter{
 			Next:   next,
 			Config: ifconfig,
 		}
-	}, nil
+	}
+	// Add middleware
+	cfg := httpserver.GetConfig(c.Key)
```
Comment (pyed) [inline ipfilter.go]: should be `httpserver.GetConfig(c)`

File: ipfilter_test.go [hunk 2]
```diff
@@ -757,7 +757,7 @@ func TestIpfilterParse(t *testing.T) {
 	}
 
 	for i, test := range tests {
-		c := setup.NewTestController(test.inputIpfilterConfig)
+		c := caddy.NewTestController(test.inputIpfilterConfig)
```
Comment (pyed) [inline ipfilter_test.go]: should be `caddy.NewTestController("http", test.inputIpfilterConfig)`

**Pull request merged**